src/vs/platform/terminal/common/capabilities/commandDetectionCapability.ts

```diff
@@ -380,7 +386,7 @@ export class CommandDetectionCapability extends Disposable implements ICommandDe
 			let i = 0;
 			for (; i < 20; i++) {
 				await timeout(10);
-				if (!this._windowsPromptPollingInProcess || this._cursorOnNextLine() && this._cursorLineLooksLikeWindowsPrompt()) {
+				if (this._isDisposed || !this._windowsPromptPollingInProcess || this._cursorOnNextLine() && this._cursorLineLooksLikeWindowsPrompt()) {
```

Comment:
Canceling via a CancelablePromise or something instead of waiting for the polling to finish would be better than this.

Reply:
This does not wait for the polling - it breaks when it's disposed of.

Reply:
the issue w the cancelable promise is we don't have an event from which to cancel the promise. we could add one in the override `dispose`, but that seems hacky.

Reply:
I guess we could also cancel the promise from within the `override dispose` function instead